searchlib/src/tests/fef/termmatchdatamerger/termmatchdatamerger_test.cpp

```diff
@@ -245,5 +245,98 @@ TEST(TermMatchDataMergerTest, merge_max_element_length)
     EXPECT_EQ(1000u, out.getIterator().getFieldLength());
 }
 
+TEST(TermMatchDataMergerTest, merge_no_normal_features)
+{
+    TermFieldMatchData a;
+    TermFieldMatchData b;
+    MDMIs input;
+    input.push_back(MDMI(&a, 0.5));
+    input.push_back(MDMI(&b, 1.5));
+
+    TermFieldMatchData out;
+    TermFieldMatchDataArray output;
+    output.add(&out);
+    out.setNeedNormalFeatures(false);
+    TermMatchDataMerger merger(input, output);
+
+    uint32_t docid = 5;
+
+    a.reset(docid);
+    a.appendPosition(make_pos(5));
+
+    b.reset(docid);
+    b.appendPosition(make_pos(3));
+
+    merger.merge(docid);
+    EXPECT_EQ(docid, out.getDocId());
+    EXPECT_EQ(0u, out.size());
+}
+
+TEST(TermMatchDataMergerTest, merge_interleaved_features)
+{
+    TermFieldMatchData a;
+    TermFieldMatchData b;
+    MDMIs input;
+    input.push_back(MDMI(&a, 0.5));
+    input.push_back(MDMI(&b, 1.5));
+
+    TermFieldMatchData out;
+    TermFieldMatchDataArray output;
+    output.add(&out);
+    out.setNeedNormalFeatures(false);
+    out.setNeedInterleavedFeatures(true);
+    TermMatchDataMerger merger(input, output);
+
+    uint32_t docid = 5;
+
+    a.reset(docid);
+    a.setNumOccs(1);
+    a.setFieldLength(30);
+
+    b.reset(docid);
+    b.setNumOccs(1);
+    b.setFieldLength(30);
```

Comment:
Consider using different value than _**a**_ so that max of field lengths are tested.